### PR TITLE
WSTEAM1-47 - Youtube embed consent banner for Optimo articles

### PR DESCRIPTION
**Pull request by amoore108**
Resolves WSTEAM1-47

**Overall change:**
Adds a consent banner to YouTube embeds on Optimo article pages. This shows each time to the user, which they must 'accept' before the embed is rendered on the page. This applies to both canonical and AMP YouTube embeds for Optimo article pages

The component has been designed to be extendable outside of just YouTube embeds and just article pages for future use.

**Code changes:**
- Adds new `EmbedConsentBanner` components for both canonical and AMP. This wraps around the SocialEmbed component to prevent rendering of the embed until the consent banner is 'accepted'
- Adds tests and Storybook components for this new component
- Updates the `Translations` type to include a new nested object for the consent banner translations

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [ ] I have assigned this PR to the Simorgh project
- [x] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/components/EmbedConsentBanner/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    });
+
+    expect(getByTestId('consentBanner')).toBeInTheDocument();
+  });
+
+  it('should not render the banner when the user has consented', async () => {
+    let getByTestId!: (id: string) => HTMLElement;
+    let queryByTestId!: (id: string) => HTMLElement | null;
+
+    await act(async () => {
```
Comment (jroebu14) [inline src/app/components/EmbedConsentBanner/index.test.tsx]: Just so know, I have a PR where the ThemeProvider is mocked to load themes synchronously. When that is merged you won't need to await like this 🙂

**Review by jroebu14 — APPROVED** [review]
Looks great

File: src/app/components/EmbedConsentBanner/index.test.tsx (hunk 1 above)
Comment (amoore108) [reply src/app/components/EmbedConsentBanner/index.test.tsx]: Ah nice! Should mean that I can remove the non-null assertion `!` as well 🙌🏼

**Review by eagerterrier — APPROVED** [review]
Swedish_Chef

**Review by jroebu14 — APPROVED** [review]

File: src/app/lib/config/services/afrique.ts [hunk 2]
```diff
@@ -258,6 +258,16 @@ export const service: DefaultServiceConfig = {
           text: 'Ignorer %provider_name% publication',
           endTextVisuallyHidden: 'Fin de %provider_name% publication',
         },
+        consentBanner: {
+          heading: 'Autoriser le contenu de [social_media_site]?',
+          body: `This article contains content provided by [social_media_site].  We ask for your permission before anything is loaded, as they may be using cookies and other technologies.  You may want to read the [social_media_site] [link] cookie policy [/link] and [link] privacy policy [/link] before accepting. To view this content choose 'accept and continue'.`,
```
Comment (HarveyPeachey) [inline src/app/lib/config/services/afrique.ts:263]: No translations for afrique?

File: src/app/components/EmbedConsentBanner/ConsentBanner.styles.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import { css, Theme } from '@emotion/react';
+
+export default {
+  self: ({ spacings, palette }: Theme) =>
```
Comment (HarveyPeachey) [inline src/app/components/EmbedConsentBanner/ConsentBanner.styles.tsx]: Is self meant to be the top-level element?

**Comment by HarveyPeachey** [comment]
Is it in scope for users accepting the banner, that their choice is saved in a cookie so they don't have subsequently accept the banner if they navigate to a new article with another youtube embed?

File: src/app/lib/config/services/afrique.ts (hunk 2 above)
Comment (amoore108) [reply src/app/lib/config/services/afrique.ts:263]: There are a few translations missing for the body text and 'Accept and Continue' button. @JonBeeb said to default to English in these cases.

File: src/app/components/EmbedConsentBanner/ConsentBanner.styles.tsx (hunk 3 above)
Comment (amoore108) [reply src/app/components/EmbedConsentBanner/ConsentBanner.styles.tsx]: I've updated this to be `parent`, as it is the top-level wrapping element for the banner

**Comment by amoore108** [comment]
> Is it in scope for users accepting the banner, that their choice is saved in a cookie so they don't have subsequently accept the banner if they navigate to a new article with another youtube embed?

One for @JonBeeb perhaps? I vaguely remember a conversation that we wouldn't do this. The acceptance criteria says that the banner should display for each embed if there are multiple on the page, so given that I would think it would appear each time there is an embed regardless if they clicked 'Accept'.

**Review by HarveyPeachey — APPROVED** [review]

**Comment by JonBeeb** [comment]
Hello, so this version is just show every time on every embed. There (may) be future iterations where we somehow store the users choice...

File: src/app/components/EmbedConsentBanner/ConsentBanner.styles.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+  textBody: ({ spacings, palette }: Theme) =>
+    css({
+      color: palette.GREY_10,
+      margin: `${spacings.DOUBLE}rem 0`,
+
+      a: {
+        color: 'inherit',
+        textDecoration: 'none',
+        borderBottom: `1px solid ${palette.GREY_10}`,
```
Comment (greenc05) [inline src/app/components/EmbedConsentBanner/ConsentBanner.styles.tsx:28]: @amoore108 we usually use rem even for borders (same for hover, focus and visisted)

File: src/app/components/EmbedConsentBanner/ConsentBanner.styles.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+          borderBottom: `1px solid ${palette.GREY_6}`,
+        },
+      },
+    }),
+
+  button: ({ spacings, palette }: Theme) =>
+    css({
+      color: palette.GREY_10,
+      backgroundColor: palette.WHITE,
+      border: `1px solid ${palette.PHILIPPINE_GREY}`,
```
Comment (greenc05) [inline src/app/components/EmbedConsentBanner/ConsentBanner.styles.tsx:46]: @amoore108 we usually use rem even for borders (same for the button hover states etc)

**Pull request merged**